test/development/app-dir/owner-stack/owner-stack.test.ts

```diff
@@ -1,112 +1,83 @@
 import { nextTestSetup } from 'e2e-utils'
-import {
-  assertHasRedbox,
-  assertNoRedbox,
-  openRedbox,
-  getRedboxDescription,
-  hasRedboxCallStack,
-} from 'next-test-utils'
-
-// TODO: parse the location and assert them in the future
+import { assertNoRedbox, retry } from 'next-test-utils'
+
 // Remove the location `()` part in every line of stack trace;
 // Remove the leading spaces in every line of stack trace;
 // Remove the trailing spaces in every line of stack trace;
-function normalizeStackTrace(trace: string) {
+// These stacks are not sourcemapped and therefore not ignore-listed.
+// Feel free to update internal frames in assertions.
+function normalizeBrowserConsoleStackTrace(trace: unknown) {
+  if (typeof trace !== 'string') {
+    return trace
+  }
   return trace
     .replace(/\(.*\)/g, '')
     .replace(/^\s+/gm, '')
     .trim()
 }
 
-async function getStackFramesContent(browser) {
-  await hasRedboxCallStack(browser)
-  const stackFrameElements = await browser.elementsByCss(
-    '[data-nextjs-call-stack-frame]'
-  )
-  const stackFramesContent = (
-    await Promise.all(
-      stackFrameElements.map(async (frame) => {
-        const functionNameEl = await frame.$('[data-nextjs-frame-expanded]')
-        const sourceEl = await frame.$('[data-has-source]')
-        const functionName = functionNameEl
-          ? await functionNameEl.innerText()
-          : ''
-        const source = sourceEl ? await sourceEl.innerText() : ''
-
-        if (!functionName) {
-          return ''
-        }
-        return `at ${functionName} (${source})`
-      })
-    )
-  )
-    .filter(Boolean)
-    .join('\n')
-
-  return stackFramesContent
-}
-
 describe('app-dir - owner-stack', () => {
-  const { next } = nextTestSetup({
+  const { isTurbopack, next } = nextTestSetup({
     files: __dirname,
   })
 
   it('should log stitched error for browser uncaught errors', async () => {
-    const browser = await next.browser('/browser/uncaught')
-
-    await assertHasRedbox(browser)
+    let errorStack: string | undefined
+    const browser = await next.browser('/browser/uncaught', {
+      beforePageLoad: (page) => {
+        page.on('pageerror', (error: unknown) => {
+          errorStack = (error as any).stack
+        })
+      },
+    })
+
+    if (!isTurbopack) {
+      // Wait for Redbox to settle.
+      // TODO: Don't reload when landing on a faulty page.
+      // The issue may be that we receive an HMR update at all on landing.
+      // This is flaky. Sometimes we miss that the reload happened.
+      await retry(
+        async () => {
+          const logs = await browser.log()
+          expect(logs).toEqual(
+            expect.arrayContaining([
+              expect.objectContaining({
+                message:
+                  '[Fast Refresh] performing full reload because your application had an unrecoverable error',
```

Comment:
usually when it's happening it might be a bug, but I couldn't repro this

Reply:
This is intentional. Otherwise your screen wouldn't update if you make an edit after an error boundary caught an error. I can repro this on a fresh dev server.